lindi/LindiH5pyFile/LindiH5pyFile.py

```diff
@@ -14,19 +14,35 @@
 
 
 class LindiH5pyFile(h5py.File):
-    def __init__(self, _file_object: Union[h5py.File, zarr.Group]):
+    def __init__(self, _file_object: Union[h5py.File, zarr.Group], *, _zarr_store: Union[ZarrStore, None] = None, _mode: Literal["r", "r+"] = "r"):
         """
         Do not use this constructor directly. Instead, use:
         from_reference_file_system, from_zarr_store, from_zarr_group,
         or from_h5py_file
         """
         self._file_object = _file_object
+        self._zarr_store = _zarr_store
+        self._mode: Literal['r', 'r+'] = _mode
         self._the_group = LindiH5pyGroup(_file_object, self)
 
     @staticmethod
-    def from_reference_file_system(rfs: Union[dict, str]):
+    def from_reference_file_system(rfs: Union[dict, str], mode: Literal["r", "r+"] = "r"):
         """
         Create a LindiH5pyFile from a reference file system.
```

Comment:
Would any RFS JSON, e.g., one generated by `kerchunk` work here? It should be fine, right?

Reply:
yes it would.